routers/api/v1/repo/tree.go

```diff
@@ -87,29 +102,40 @@ func GetTreeBySHA(ctx *context.APIContext, sha string) *gitea.GitTreeResponse {
 	// 40 is the size of the sha1 hash in hexadecimal format.
 	copyPos := len(treeURL) - 40
 
-	if len(entries) > 1000 {
-		tree.Entries = make([]gitea.GitEntry, 1000)
+	page := ctx.QueryInt("page")
+	perPage := ctx.QueryInt("per_page")
+	if perPage <= 0 {
```

Comment:
I think we still should not allow more than 1K entries
```suggestion
	if perPage <= 0 || perPage > setting.API.DefaultGitTreesPerPage {
```